pkg/configOperator/configOperator.go

```diff
@@ -106,7 +106,7 @@ func (op ConfigOperator) handleDeleteSecret(obj runtime.Object) {
 // If a client config is found that references a variant not found in UPS then we clean up the client config by deleting the associated servicebinding.
 func (op ConfigOperator) compareUPSVariantsWithClientConfigs() {
 	// get the UPS related secrets
-	selector := fmt.Sprintf("serviceName=ups,pushApplicationId=%s", op.pushClient.getApplicationId())
+	selector := fmt.Sprintf("serviceName=ups,pushApplicationId=%s", op.pushClientProvider.getPushClient().getApplicationId())
```

Comment:
This code can run before the secret exists. How would this work when `pushClientProvider.getPushClient()` still returns nil or throws here?

Reply:
`op.pushClientProvider.getPushClient().getApplicationId()` will log something and then  throw an error. can't tell for sure what the ideal situation should be

Reply:
I think it would need something like:

```
client, err := op.pushClientProvider.getPushClient()

if err != nil {
  log.warn("push client not yet available")
  return
}

```

Reply:
ah, got what you mean now @pb82 . it will throw an error and the program will be dead :(